core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/LongTerms.java

```diff
@@ -152,4 +154,52 @@ public XContentBuilder doXContentBody(XContentBuilder builder, Params params) th
     protected Bucket[] createBucketsArray(int size) {
         return new Bucket[size];
     }
+
+    @Override
+    public InternalAggregation doReduce(List<InternalAggregation> aggregations, ReduceContext reduceContext) {
+        boolean promoteLongsToDouble = false;
+        DocValueFormat decimalFormat = null;
+        for (InternalAggregation agg : aggregations) {
+            if (agg instanceof DoubleTerms) {
+                /**
+                 * this terms agg mixes longs and doubles, we must promote longs to doubles to make the internal aggs
+                 * compatible
+                 */
+                promoteLongsToDouble = true;
+                decimalFormat = ((DoubleTerms) agg).format;
+                break;
```

Comment:
Maybe in that case we could just `return agg.doReduce` on this DoubleTerms instance which would reuse the logic we have in DoubleTerms rather than duplicating it below?